ha/core/system_health/health_evaluators/element_health_evaluator.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import abc
+import json
+import re
+
+from cortx.utils.log import Log
+from ha.core.system_health.system_health_hierarchy import HealthHierarchy
+from ha.core.config.config_manager import ConfigManager
+from ha.core.system_health.model.health_event import HealthEvent
+from ha.core.system_health.system_health_metadata import SystemHealthComponents
+from ha.core.error import HaSystemHealthException
+from ha.core.system_health.system_health_manager import SystemHealthManager
+from ha.core.system_health.system_health_exception import HealthNotFoundException
+
+# TODO: (Task Refactore) Refactore and move system health update element code to element
+
+class ElementHealthEvaluator(metaclass=abc.ABCMeta):
+    """
+    Generic cluster elements class.
+    """
+
+    HEALTH_MANAGER = None
+
+    def __init__(self):
+        self._confstore = ConfigManager.get_confstore()
```

Comment:
SystemHealth gets a store when it is instantiated, pass it here and use.

Reply:
want to keep `ElementHealthEvaluator`  parameter less

i added one TODO to make `SystemHealthManager` as singleton class
which will resolve this issue
it will be like SystemHealthManager initalized by systemhealth and 

ElementHealthEvaluator can get instance by 
```
self.healthmanager = SystemHealthManager.get_instance()
self.healthmanager.get_store() // if needed to get store
```

Reply:
Sure, please create one improvement story and add all the TODOs from this PR in it.

Reply:
Sure will create new ticket